### PR TITLE
HOT FIX: allow latency mode (streams = 1) to use hyper threading processors

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *allow latency mode (streams = 1) to use hyper threading processors*

### Tickets:
 - *CVS-107363*

### CODE REVIEW

**Comment by wangleis** [comment]
@dmitry-gorokhov This is a urgent requirement. Please take a look.

**Comment by dmitry-gorokhov** [comment]
@sunxiaoxia2022 @peterchen-intel @xufang-lisa guys, could you please review from the implementation details perspective?

**Comment by dmitry-gorokhov** [comment]
@wangleis Do I correctly understand that combination of hint=latency and enable_hyper_thread=true hadn't actually led to hyperthreads usage during compuation? 
If so can we analyse existing unit test instances it terms of what combinations are not covered yet asuming the following params: hint, nstreams, nthreads, hyper-threads, pinning, core-types (smt else?).

**Comment by wangleis** [comment]
@dmitry-gorokhov yes, your understanding is right. In current implementation,  "latency mode only use physical cores" is higher priority.

So far smoke test for multi threading 2.0 just cover independent key features. Optimization and smoke test for mix of configuration and streams scheduling was planned.

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp [hunk 1]
```diff
@@ -30,48 +30,68 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
...
-            stream_info[PROC_TYPE] = MAIN_CORE_PROC;
-            stream_info[THREADS_PER_STREAM] = (limit_threads == 0)
-                                                  ? proc_type_table[0][MAIN_CORE_PROC]
-                                                  : std::min(proc_type_table[0][MAIN_CORE_PROC], limit_threads);
-            streams_info_table.push_back(stream_info);
+            if (proc_type_table[0][ALL_PROC] == proc_type_table[0][EFFICIENT_CORE_PROC]) {
+                stream_info[PROC_TYPE] = EFFICIENT_CORE_PROC;
+                stream_info[THREADS_PER_STREAM] =
+                    (input_threads == 0) ? proc_type_table[0][EFFICIENT_CORE_PROC]
+                                         : std::min(proc_type_table[0][EFFICIENT_CORE_PROC], model_prefer_threads);
```
Comment (xufang-lisa) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:74]: It looks like the input_threads in this branch should be 0, so here condition may be "model_prefer_threads == 0"?

Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:74]: Thanks. Updated.

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp [hunk 2]
```diff
@@ -30,48 +30,68 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
...
-                }
+        if (input_threads > 0) {
+            stream_info[THREADS_PER_STREAM] = std::min(proc_type_table[0][ALL_PROC], input_threads);
+            if ((stream_info[THREADS_PER_STREAM] > proc_type_table[0][MAIN_CORE_PROC]) &&
+                (proc_type_table[0][MAIN_CORE_PROC] > 0) && (proc_type_table[0][EFFICIENT_CORE_PROC] > 0)) {
+                stream_info[PROC_TYPE] = ALL_PROC;
+                streams_info_table.push_back(stream_info);
+                UpdateMixStreamInfo();
+            } else if ((stream_info[THREADS_PER_STREAM] <= proc_type_table[0][MAIN_CORE_PROC]) ||
+                       (proc_type_table[0][EFFICIENT_CORE_PROC] == 0)) {
```
Comment (sunxiaoxia2022) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:62]: Is there such a situation as `stream_info[THREADS_PER_STREAM] > proc_type_table[0][MAIN_CORE_PROC] && proc_type_table[0][EFFICIENT_CORE_PROC] == 0` which should uses logical core?

Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:62]: Yes. For one stream using both physical core and logic core of Pcore, the core type of stream will be physical core of Pcore, and number of threads in stream will be total number of physical core and logic core.

**Review by xufang-lisa — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Review by sunxiaoxia2022 — APPROVED** [review]

**Comment by wangleis** [comment]
@dmitry-gorokhov Now this PR can support  hint=latency and enable_hyper_thread=true to use all CPU resource.

**Review by riverlijunjie — APPROVED** [review]

**Pull request merged**